parquet/src/main/java/org/apache/iceberg/parquet/TypeWithSchemaVisitor.java

```diff
@@ -211,13 +211,13 @@ private static <T> List<T> visitFields(
   }
 
   private static <T> T visitVariant(
-      Types.VariantType variant, GroupType group, TypeWithSchemaVisitor<T> visitor) {
+      Types.VariantType variant, GroupType variantGroup, TypeWithSchemaVisitor<T> visitor) {
```

Comment:
In order to call a visitor that has a different return type than the Parquet schema visitor (in this case, `VariantMetrics` instead of `FieldMetrics`), the Parquet type needs to be passed to the visit method. I think that not including this originally was an accident.